pkg/policy/rule.go

```diff
@@ -58,19 +58,44 @@ func (r *rule) String() string {
 	return fmt.Sprintf("%v", r.EndpointSelector)
 }
 
-func mergePortProto(ctx *SearchContext, endpoints []api.EndpointSelector, existingFilter, filterToMerge *L4Filter) error {
+func mergePortProto(ctx *SearchContext, existingFilter, filterToMerge *L4Filter, selectorCache *SelectorCache) error {
```

Comment:
Even though this function is not exported, I think it'd be beneficial to add unit tests for this function due to the high importance of how we manage objects in the `SelectorCache` here.